src/pages/VenuePage/VenuePage.tsx

```diff
@@ -252,6 +258,55 @@ const VenuePage = () => {
     }
   }, [user, profile, venueId, venueTemplate, mixpanel]);
 
+  // Need a token to access the venue
+  useEffect(() => {
+    if (!venue) return;
+
+    const denyAccess = () => {
+      localStorage.removeItem(getAccessTokenKey(venueId));
+      setIsAccessDenied(true);
+    };
+
+    // @debt convert this so token is needed to get venue config
```

Comment:
> `@debt` convert this so token is needed to get venue config

Reply:
The code this @debt was attached to was refactored into `src/hooks/useVenueAccessToken.ts` in https://github.com/sparkletown/sparkle/pull/1016/commits/3a4a3d7ec1e2f6519931827554f4c7b79bf735cf

---

@denisdimitrov Can you please explain how the moved code addresses the `@debt` called out here? It seems (at a quick skim) to be the same code that existed before, just encapsulated in the custom hook now?

Reply:
I have no idea, the code wasn't written by me and I have no idea what the intention was. By the looks of it, the token is already being used before checking for the venue access. But I can't quite understand the idea with only so much context. Venue config is being used in ~40 places. Maybe Chris can elaborate here.

Reply:
@denisdimitrov I wasn't sure if this was related to changes you had made to this PR or if it happened prior to you inheriting it. Given you've more recently been involved with it than me, I figured that you may have had more updated knowledge to figure it out. If not, all good.

@thecadams any insight on this?